src/commands/transaction/sign.ts

```diff
@@ -99,7 +140,20 @@ export default class SignCommand extends BaseIPCCommand {
 		transactionObject.asset = codec.fromJSON(assetSchema.schema, asset);
 		let signedTransaction: Record<string, unknown>;
 
-		if (mandatoryKeys || optionalKeys) {
+		// sign from multi sig account offline using input keys
+		if (!includeSender && !senderPublicKey) {
+			signedTransaction = transactions.signTransaction(
+				assetSchema.schema,
+				transactionObject,
+				networkIdentifierBuffer,
+				passphrase,
+			);
+		} else if (offline) {
+			if (!mandatoryKeys.length || !optionalKeys.length) {
+				throw new Error(
+					'--mandatory-keys or --optional-keys flag must be specified to sign transaction from multi signature account.',
```

Comment:
both are mandatory, right?, also, i have a question, optionalKeys could be zero as well right?
```suggestion
					'--mandatory-keys and --optional-keys flag must be specified to sign a transaction from the multi-signature account.',
```

Reply:
user can create with only mandatory or optional.. so its either of the one.

Reply:
but then the above condition will fail if there is no optional as the condition required it to have both,
`!mandatoryKeys.length || !optionalKeys.length` it should be `!mandatoryKeys.length && !optionalKeys.length` if both are empty then it should throw error